code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm

```diff
@@ -0,0 +1,448 @@
+	. = ..()
+	var/list/data = list()
+
+	data["output"] = output_level
+	data["total_earnings"] = total_earnings
+	data["unsent_earnings"] = unsent_earnings
+	data["held_power"] = charge
+	data["selling_energy"] = selling_energy
+	data["max_capacity"] = capacity
+	data["max_grid_load"] = max_grid_load
+
+	data["accepting_power"] = turned_on
+	data["sucking_power"] = inputting
+	data["firing"] = firing
+
+	data["power_format"] = power_format_multi
+	data["input_number"] = input_number
+	data["avalible_input"] = input_available
+	data["output_number"] = output_number
+	data["output_multiplier"] = power_format_multi_output
+	data["input_total"] = input_number * power_format_multi
+	data["output_total"] = output_number * power_format_multi_output
+
+	return data
+
+/obj/machinery/power/transmission_laser/ui_act(action, list/params)
+	. = ..()
+	if(.)
+		return
+	switch(action)
+		if("toggle_input")
+			turned_on = !turned_on
+			update_icon()
+		if("toggle_output")
+			firing = !firing
+			if(!firing)
+				destroy_lasers()
+			update_icon()
+
+		if("set_input")
+			input_number = clamp(params["set_input"], 0, 999) //multiplies our input by if input
+		if("set_output")
+			output_number = clamp(params["set_output"], 0, 999)
+
+		if("inputW")
+			power_format_multi = 1
+		if("inputKW")
+			power_format_multi = 1 KW
+		if("inputMW")
+			power_format_multi = 1 MW
+		if("inputGW")
+			power_format_multi = 1 GW
+
+		if("outputW")
+			power_format_multi_output = 1
+		if("outputKW")
+			power_format_multi_output = 1 KW
+		if("outputMW")
+			power_format_multi_output = 1 MW
+		if("outputGW")
+			power_format_multi_output = 1 GW
+
+/obj/machinery/power/transmission_laser/process()
+	max_grid_load = get_surplus()
+	input_available = get_surplus()
+	if(stat & BROKEN)
+		return
+
+	if(total_energy >= announcement_threshold)
+		send_ptl_announcement()
+
+	var/last_disp = return_charge()
+	var/last_chrg = inputting
+	var/last_fire = firing
+
+	if(last_disp != return_charge() || last_chrg != inputting || last_fire != firing)
+		update_icon()
+
+	if(powernet && input_attempt && turned_on)
+		input_pulling = min(input_available , input_number * power_format_multi)
+
+		if(inputting)
+			if(input_pulling > 0)
+				consume_direct_power(input_pulling)
+				charge += input_pulling
+			else
+				inputting = FALSE
+		else
+			if(input_attempt && input_pulling > 0)
+				inputting = TRUE
+	else
+		inputting = FALSE
+
+	if(charge < MINIMUM_POWER)
+		firing = FALSE
+		output_level = 0
+		destroy_lasers()
+		return
+
+	if(!firing)
+		return
+
+	output_level = min(charge, output_number * power_format_multi_output)
+	if(!length(laser_effects))
+		setup_lasers()
+
+	if(length(blocked_objects))
+		var/atom/listed_atom = blocked_objects[1]
+		if(prob(max((abs(output_level) * 0.1) / 500 KW, 1))) ///increased by a single % per 500 KW
+			listed_atom.visible_message("<span class='danger'>[listed_atom] is melted away by the [src]!<span/>")
```

Comment:
This should probably have special handling for mobs. Deal burn damage, dust them, whatever. Just don't `qdel()` them with no recourse.

Reply:
Mob handling is done in atom_entered_beam. blocked_objects should contain turfs and objects. Looking at it though, it seems like objects are never added to it. that's odd. I'll have to test that and fix what's wrong.

Reply:
I tested and it is indeed broken. It probably should have been done when setting up the lasers, but it's simply ignored. I looked around monkey station and it doesn't look like the code for it is there either. 
I'll fix this.

Reply:
Sorted this